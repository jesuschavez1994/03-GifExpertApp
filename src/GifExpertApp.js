import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ( { deafultCategories = [] } ) => {

    const [categories, setCategories] = useState( deafultCategories );

    // const handleAdd = (e) => setCategories( [...categories, 'Pokemon'] );

    return (
        <>
            <h2>GifExpert</h2>
                <AddCategory  setCategories = { setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category => {
                        return <GifGrid category = { category } key={category} />
                    })
                }
            </ol>
        </>
    )

}

export default GifExpertApp;