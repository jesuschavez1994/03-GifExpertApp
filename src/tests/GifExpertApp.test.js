import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';


describe('Pruebas en zGifExpertApp />', () => {

    let wrapper = shallow(<GifExpertApp />);

    test('debe de mostrar correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar una lista de elementos ', () => {

        const categories = ['One punch', 'Naruto'];
        let wrapper = shallow(<GifExpertApp  deafultCategories = { categories }/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
       
    })
    
    
})
