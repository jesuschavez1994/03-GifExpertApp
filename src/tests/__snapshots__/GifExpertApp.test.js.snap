// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en zGifExpertApp /> debe de mostrar correctamente  1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en zGifExpertApp /> debe de mostrar una lista de elementos  1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;
